yt/data_objects/static_output.py

```diff
@@ -871,47 +871,41 @@ def _add_object_class(self, name, base):
         obj.__doc__ = base.__doc__
         setattr(self, name, obj)
 
-    def find_max(self, field):
+    def _find_extremum(self, field, ext, source=None, to_array=True):
+        ext = ext.lower()
+        if source is None:
+            source = self.all_data()
+        method = {
+            "min": source.quantities.min_location,
+            "max": source.quantities.max_location,
+        }[ext]
+        val, x1, x2, x3 = method(field)
+        coords = [x1, x2, x3]
+        mylog.info("%s value is %0.5e at %0.16f %0.16f %0.16f", ext, val, *coords)
+        if to_array:
+            # This is a hack to fix the fact that some non-cartesian datasets have
+            # dimensionless quantities, and we can't yet handle that.
+            alt_coords = []
+            for x in coords:
+                alt_coords.append(
+                    self.quan(x.v, "code_length") if x.units.is_dimensionless else x
+                )
+            coords = self.arr(alt_coords, dtype="float64").to("code_length")
```

Comment:
This looks a bit bizarre to me. If a user calls the method with `to_array` in non-cartesian coordinates, why don't we throw an error or issue a warning and disable `to_array`?

Reply:
Oh, I just read the deleted code and realize this was the already-existing behavior. Anyways, I would argue _against_ this, since it will silently return an array with incorrect units in the non-cartesian directions...

Reply:
I'm not a big fan either. Maybe we should throw an error instead (or maybe just a runtime warning ?) in the non-cartesian cases and warn the user they should set `to_array=False`... wdyt ?

Reply:
I believe throwing an error encouraging to use `to_array` would be better, as most yt user I encountered do not even read the warnings anyway.

Reply:
I added an error and docstrings !

Reply:
Ah, of course this is breaking a lot of tests on Jenkins... which probably means potentially a lot of downstream code... @cphyc, @matthewturk  do you think this is worth the break ?

Reply:
If that's breaking a lot of downstream code, I'm also ok with issuing a warning.

Reply:
adressed !